sdk/keyvault/keyvault-secrets/test/README.md

```diff
@@ -1,29 +1,75 @@
+
+**Register a new application in AAD**
+
+- Follow [Documentation to register a new application](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app) in the Azure Active Directory(in the azure-portal).
+- Note down the `CLIENT_ID` and `TENANT_ID`.
+- In the "Certificates & Secrets" tab, create a secret and note that down.
+
+**Assign owner role to the registered application**
+
+- In the azure-portal, go to your keyvault-namespace and assign the **Owner** role to the registered application.
+- This can be done from `Role assignment` section of `Access control (IAM)` tab (in the left-side-navbar of your keyvault-namespace in the azure-portal)<br>
+  _Doing this would allow the registered application manage the namespace, i.e., entity creation, deletion, etc.,_<br>
+- For more information on securing your KeyVault - [Learn more](hhttps://docs.microsoft.com/en-us/azure/key-vault/general/secure-your-key-vault)
+
+**Update your environment variables**
+
+Populate the following variables along with the above mentioned environment variables in the `.env`.
+```
+AZURE_CLIENT_ID=""
+AZURE_CLIENT_SECRET=""
+AZURE_TENANT_ID=""
+```
+
+## Run all tests
+
+Run `rushx test:node` or `rushx test:browser` from your terminal to run all the tests under the `./test` folder
+
+## Run all tests in a single test suite
+
+Append the `.only` on the `describe` method corresponding to the test suite.
+
+Then run `rushx test:node` or `rushx test:browser` from your terminal.
+
+## Run a single test
+
+Append the `.only` on the `it` method corresponding to the test.
+
+Then run `rushx test:node` or `rushx test:browser` from your terminal.
+
+## Debug tests using Visual Studio Code
+
+- Select the debug configuration `Debug Unit Tests` from the drop down in the debug viewlet.
+- Add breakpoints in the test you want to debug. Tip: It is helpful to append the `.only` on the test method if you want only that method to be run.
+- Start Debugging
 
 ## Unit tests
 
-To run unit tests, go to the root of this library (keyvault-certificates) and run `npm run unit-test`.
-This will run tests in a purely nodejs environment, and in a browser environment using [Karma](https://karma-runner.github.io/latest/index.html).
-These tests aim to reproduce the behavior of our library against remote endpoints through previously recorded HTTP request and responses.
-You won't need to have any environment configuration, nor internet access, for these tests to run.
+To run unit tests, go to the root of this library and run `rushx unit-test`. This will run tests in a purely nodejs environment, and in a browser environment using [Karma](https://karma-runner.github.io/latest/index.html). These tests aim to reproduce the behavior of our library against remote endpoints through previously recorded HTTP request and responses. You won't need to have any environment configuration, nor internet access, for these tests to run.
```

Comment:
```suggestion
To run unit tests, go to the root of this library and run `rushx unit-test`. This will run tests in NodeJS and in a browser (through [Karma](https://karma-runner.github.io/latest/index.html)). These tests aim to reproduce the behavior of our library against remote endpoints through previously recorded HTTP request and responses. You won't need to have any environment configuration, nor internet access, for these tests to run.
```